src/Factory/Api/Friendica/Photo.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Factory\Api\Friendica;
+
+use Friendica\App\BaseURL;
+use Friendica\BaseFactory;
+use Friendica\Database\DBA;
+use Friendica\Factory\Api\Twitter\Status;
+use Friendica\Model\Photo as ModelPhoto;
+use Friendica\Model\Post;
+use Friendica\Network\HTTPException;
+use Psr\Log\LoggerInterface;
+use Friendica\Util\Images;
+
+class Photo extends BaseFactory
+{
+	/** @var BaseURL */
+	private $baseUrl;
+	/** @var Status */
+	private $status;
+	/** @var Activities */
+	private $activities;
+
+	public function __construct(LoggerInterface $logger, BaseURL $baseURL, Status $status, Activities $activities)
+	{
+		parent::__construct($logger);
+
+		$this->activities = $activities;
+		$this->status     = $status;
+		$this->baseUrl    = $baseURL;
+	}
+
+	/**
+	 * @param string $photo_id
+	 * @param int    $scale
+	 * @param int    $uid
+	 * @param string $type
+	 * @return Array
```

Comment:
So in fact we're creating a "fully functional" `Photo` object.
What about creating it as an entity `Photo`, having all the array content as type-save (readonly?) properties.
Pinging @MrPetovan as well for cross check & discussion

Reply:
Yeah, this can be improved. But at some point in time I was just happy to remove the old stuff, so I got a bit lazy.